src/main/daml/Daml/Finance/Holding/Fungible.daml

```diff
@@ -69,19 +70,21 @@ template Fungible
       asTransferable = toInterface @Transferable.I this
       view = Fungible.View with modifiers = singleton account.owner
       split Fungible.Split{amounts} = do
-        splitCids <- forA amounts \a -> toInterfaceContractId <$> create this with amount = a
+        splitCids <- forA amounts spawn
         restAmount <- getRestAmount amounts amount
-        rest <- T.forA restAmount \r -> toInterfaceContractId <$> create this with amount = r
+        rest <- T.forA restAmount spawn
         pure Fungible.SplitResult with splitCids; rest
       merge Fungible.Merge{fungibleCids} = do
+        assertMsg "list of fungibles must be non-empty" $ not $ null fungibleCids
         let
-          f a cid = do
+          aggregate a cid = do
             Some (fungibleCid, fungible) <- fetchFromInterface @Fungible cid
             assertMsg "instrument must match" $ instrument == fungible.instrument
+            assertMsg "account must match" $ account == fungible.account
```

Comment:
Shall we also check that the `typerep` of the two holdings matches?

Currently we could create a `MaliciousHolding` and merge it with the non-malicious holding. So far we have however consciously excluded such scenarios.

Reply:
As we rely on code vetting, I guess it is fine to leave it out in our implementation. 

I was considering adding checks that the account is active upon split/merge, but that adds an extra fetch, and you could argue that it should be fine to split/merge anyway. I was also considering making use of the Account's Credit/Debit instead of create/archive, but that would add even more overhead (in terms of fetches).

Reply:
As discussed with @georg-da, instead of the `typerep` check let us add 
`assertMsg "signatories must match" $ fromList (signatory this) == fromList (signatory fungible)`
to our implementation as a remedy.